### PR TITLE
doc(quality-time): add quality time meeting notes from 2024-12-11

### PR DESCRIPTION
**Pull request by myadryshnikova**

### CODE REVIEW

File: meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Кейс 2: "Второй Парень" хочу удалить (добавлено случайно)
+
+Кейс 3: Хочу добавить нового автора: "Четвертый Четвертый
+
+Кейс 4: Автора "Третий Третий" оставляем без изменений
+
+Кейс 5: Что произойдет когда удалим/переименуем автора, который: 1) есть в нескольких книгах; 2) есть только в одной книге
+
+
+## Домашнее задание
```
Comment (akovylyaeva) [inline meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md:73]: can we call this not like "Домашнее задание", but maybe just like "задание" or "надо сделать"? it`s not really important but for me it sounds a little strange;(

File: meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+## Валидация
+1. Книга не может быть без автора - проверить, что авторов в добавляемой книге > 1
+2. Определить максимальную длину строк в полях книги, автора и ограничить
+3.  Валидация должна быть на уровне базы, на уровне бэка и на уровне фронта
+
+
+## Что проверить/посмотреть?
+1. Как хранить язык книги - enum в виде строки (если это пофиксили)
+2. Доклад про B-Tree индекс
```
Comment (fpandyz) [inline meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md:11]:
```suggestion
2. [Доклад про B-Tree индекс](https://www.youtube.com/watch?v=y-Wtyvme4gE)
```

File: meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+3. Как сейчас реализовано удаление книг? Если удалить книгу, удалится ли автор? Если удалить автора, удалится ли книга?
+4. Как возвращаем сообщение об ошибке? (RFC?)
```
Comment (fpandyz) [inline meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md:13]:
```suggestion
4. Как возвращаем сообщение об ошибке https://learn.microsoft.com/en-us/aspnet/core/web-api/handle-errors?view=aspnetcore-8.0? ([RFC](https://tools.ietf.org/html/rfc7807))
```

File: meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+5. При многократном удалении одного и того же результата не возвращаем ошибку, а только { “isDeleted”: true } - идемпотентность
+6. Написать E2E на Karate для всех CRUD операций
+7. Прописать в контрактах все респонсы
+8. Реализовать редактирование для всех полей, в т.ч. автор, язык
+9. Добавить примеры значений полей в базе данных (чтобы было понятно, какие данные лежат в реальной используемой таблице - отразить в диаграмме в виде дополнительной таблицы (или если можно - в ней же, если есть такой функционал))
+10. Добавить все поля, которых не хватает в доке, но они уже есть в реализации: tenant_id, deleted_at_utc и т.д.
+
+## Каких контрактов не хватает?
+1. Запрос на одну книгу (по id)
+
```
Comment (fpandyz) [inline meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md:38]:
```suggestion
2. Поддержка тенантов согласно https://learn.microsoft.com/en-us/ef/core/miscellaneous/multitenancy, для кейса хранения колонки `TenantId` в каждой таблице.
```

File: meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+2. Определить максимальную длину строк в полях книги, автора и ограничить
+3.  Валидация должна быть на уровне базы, на уровне бэка и на уровне фронта
+
+
+## Что проверить/посмотреть?
+1. Как хранить язык книги - enum в виде строки (если это пофиксили)
+2. Доклад про B-Tree индекс
+3. Как сейчас реализовано удаление книг? Если удалить книгу, удалится ли автор? Если удалить автора, удалится ли книга?
+4. Как возвращаем сообщение об ошибке? (RFC?)
+5. Краткие коды языков для контрактов по книжкам ("eng" скорее всего может быть просто "e")
```
[the remark this answers is not in the document]
Comment (myadryshnikova) [reply meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md:14]: I think I dont understand this comment :( can you please put it another way?

File: meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md (hunk 4 above)
Comment (myadryshnikova) [reply meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md:38]: are you sure it should be in this section?

File: meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (akovylyaeva) [reply meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md:14]: in my opinion codes like "e" or "r" are unclear
more useful codes are "eng", "rus" or "en", "ru"
and we discuss that codes "en" and "ru" is probably better, but it doesn`t mentioned here

File: meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+2. Определить максимальную длину строк в полях книги, автора и ограничить
+3. Валидация должна быть на уровне базы, на уровне бэка и на уровне фронта
+
+
+## Что проверить/посмотреть?
+1. Как хранить язык книги - enum в виде строки (если это пофиксили)
+2. [Доклад про B-Tree индекс](https://www.youtube.com/watch?v=y-Wtyvme4gE)
+3. Как сейчас реализовано удаление книг? Если удалить книгу, удалится ли автор? Если удалить автора, удалится ли книга?
+4. Как возвращаем сообщение об ошибке https://learn.microsoft.com/en-us/aspnet/core/web-api/handle-errors?view=aspnetcore-8.0? ([RFC](https://tools.ietf.org/html/rfc7807))
+5. Краткие коды языков для контрактов по книжкам ("eng" скорее всего может быть просто "e")
```
Comment (myadryshnikova) [inline meeting-notes/2024-12-11-Books-DB-and-Books-CRUD-contracts.md:14]:
```suggestion
5. Посмотреть краткие общепринятые коды языков для контрактов по книжкам
Мы рассматривали при ревью два варинта:
- Russian/English
- rus/eng

Во время quality time вспомнили, что возможно есть еще вариант с ru/en. Нужно посмотреть, действительно ли коды языков выглядят так.
```

**Pull request merged**